### PR TITLE
`tmpnet`: Break config.go up into coherent parts

### PR DESCRIPTION
**Pull request by marun**
Largely mechanical reorganization to improve maintainability:

- Move NetworkConfig functionality to Network
- Move NodeConfig functionality to Node
- Move FlagsMap to flags.go
- Move genesis helpers to genesis.go
- Move NodeURI and key generation to utils.go
- Remove redundant node_test.go

### CODE REVIEW

File: tests/fixture/tmpnet/defaults.go [hunk 1]
```diff
@@ -19,6 +19,15 @@ const (
 	DefaultNetworkStartTimeout = 2 * time.Minute
 	DefaultNodeInitTimeout     = 10 * time.Second
 	DefaultNodeStopTimeout     = 5 * time.Second
+
+	// Minimum required to ensure connectivity-based health checks will pass
+	DefaultNodeCount = 2
```
Comment (abi87) [inline tests/fixture/tmpnet/defaults.go:24]: probably out of scope here but should this be MinNodeCount? And should we assert that we never have a single node network? Not sure if the latter is too restrictive for our testing.

File: tests/fixture/tmpnet/utils.go [hunk 2]
```diff
@@ -42,7 +45,41 @@ func WaitForHealthy(ctx context.Context, node *Node) error {
 	}
 }
 
+// NodeURI associates a node ID with its API URI.
+type NodeURI struct {
+	NodeID ids.NodeID
+	URI    string
+}
+
+func GetNodeURIs(nodes []*Node) []NodeURI {
```
Comment (abi87) [inline tests/fixture/tmpnet/utils.go:54]: q: why are these not in node.go files?

File: tests/fixture/tmpnet/node_test.go [hunk 3]
```diff
@@ -1,24 +0,0 @@
...
-
-package tmpnet
-
-import (
-	"testing"
-
-	"github.com/stretchr/testify/require"
-)
-
-func TestNodeSerialization(t *testing.T) {
```
Comment (abi87) [inline tests/fixture/tmpnet/node_test.go:12]: q: why dropping this test? IIUC it is to ensure that some default configuration/settings gives coherent results. If so I find them useful for maintenance

File: tests/fixture/tmpnet/network.go [hunk 4]
```diff
@@ -84,9 +82,13 @@ func FindNextNetworkID(rootDir string) (uint32, string, error) {
 
 // Defines the configuration required for a tempoary network
 type Network struct {
-	NetworkConfig
 	NodeRuntimeConfig
 
+	Genesis       *genesis.UnparsedConfig
+	CChainConfig  FlagsMap
+	DefaultFlags  FlagsMap
```
Comment (abi87) [inline tests/fixture/tmpnet/network.go:89]: nit: can we find a better name for `DefaultFlags`? Are these `DefaultNetworkConfigs` or `DefaultGlobalConfigs` (they seems to hold both network and node configs)

**Review by abi87 — APPROVED** [review]
just nits and questions

**Review by danlaine — APPROVED** [review]

**Pull request merged**